### PR TITLE
[limshaoqi] iP

### PR DESCRIPTION
**Pull request by limshaoqi**
# ElChino - Un Gestor de Tareas Simple 📝
> "Friends can come and go, but banners hang forever." - Kobe Bryant [source](https://www.espn.com/blog/los-angeles/lakers/post/_/id/41673/kobe-friends-can-come-and-go-but-banners-hang-forever)

## Características
- Gestión de tareas basada en comandos
- Fácil de aprender
- Respetuoso del medio ambiente

## Todo lo que necessitas hacer es:
1. Descárgalo desde aquí.
2. Haz doble clic en él.
3. Añade tus tareas.
4. Deja que administre tus tareas por ti. 😉

### RELAK LAH
- [x] baik uh sial
- [ ] *relak*
- [ ] __rizz__ & ~~chill~~

```
"Deng bu dao tian hei, Yan huo bu hui tai wan mei.
Hui yi shao cheng hui, Hai shi deng bu dao jie wei.
Ta ceng shuo de wu suo wei, Wo pa yi tian yi tian bei cui hui - Rizzalations 24:7
```

### CODE REVIEW

File: src/main/java/elchino/Elchino.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+import elchino.ui.Ui;
+
+import elchino.tasks.*;
+
+import elchino.storage.Storage;
+
+import elchino.parser.Parser;
+
+import elchino.commands.*;
```
Comment (Frozennfishh) [inline src/main/java/elchino/Elchino.java]: Perhaps the import files of the same type could be group together to be able to more easily check?

File: src/main/java/elchino/data/tasks.txt
Comment (Frozennfishh) [inline src/main/java/elchino/data/tasks.txt:1]: Maybe can remove redundant empty files like this

File: src/main/java/elchino/tasks/Task.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+     * @return The LocalDateTime object representing the date.
+     * throws IllegalArgumentException If the date string is invalid.
+     */
+    protected static LocalDateTime parseDate(String date) {
+        try {
+            return LocalDateTime.parse(date, INPUT_DATE_FORMAT);
+        } catch (DateTimeParseException e) {
+            throw new IllegalArgumentException("Fecha inválida. Formato esperado: dd/MM/yyyy HHmm");
+        }
+    }
```
Comment (Frozennfishh) [inline src/main/java/elchino/tasks/Task.java:61]: Maybe this method could be under parser instead for more clarity? 

Will there be more date formats that could be made available for future versions?

File: src/main/java/elchino/ui/Ui.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        return scanner.nextLine().trim();
+    }
+
+    /**
+     * Prints a message.
+     * @param message The message to print.
+     */
+    public void printMessage(String message) {
+        System.out.println(message);
+    }
```
Comment (Frozennfishh) [inline src/main/java/elchino/ui/Ui.java:58]: Is this necessary?

File: src/main/java/elchino/tasks/Task.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import java.time.LocalDateTime;
+
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents an abstract task that can be done.
+ */
+public abstract class Task {
+    protected final String description;
+    protected boolean isDone;
```
Comment (Frozennfishh) [inline src/main/java/elchino/tasks/Task.java:12]: Maybe Class type could be included as a var of parent class Task for all tasks for more flexibility in the future?

File: src/main/java/elchino/tasks/Task.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        } catch (DateTimeParseException e) {
+            throw new IllegalArgumentException("Fecha inválida. Formato esperado: dd/MM/yyyy HHmm");
+        }
+    }
+
+    /**
+     * Retrieves the task type identifier.
+     * @return The task type identifier.
+     */
+    public abstract String getTaskType();
```
Comment (Frozennfishh) [inline src/main/java/elchino/tasks/Task.java:67]: Referring to above comment, then this can read data from the class instead of printing manually, and may be easier to manage if down the line there may be methods that require reading class type (i.e. sort by task type)

Or a use of instanceOf to keep track of class data?

File: src/main/java/elchino/parser/Parser.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
+package elchino.parser;
+
+import elchino.exceptions.*;
+import elchino.commands.*;
```
Comment (Frozennfishh) [inline src/main/java/elchino/parser/Parser.java:4]: Perhaps be more consistent across all files where the line breaks are when displaying packages and imports

File: src/main/java/elchino/storage/Storage.java [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+                    return event;
+                default:
+                    System.out.println("Error: Tipo de tarea desconocido en el archivo: " + line);
+                    return null;
+            }
+        } catch (Exception e) {
+            System.out.println("Error: No se pudo procesar la tarea en el archivo: " + line);
+            return null;
+        }
+    }
```
Comment (Frozennfishh) [inline src/main/java/elchino/storage/Storage.java]: Should this be under Parser as well?

File: src/main/java/elchino/tasks/TaskList.java [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+    public void printTasks() {
+        if (tasks.isEmpty()) {
+            System.out.println("No hay tareas.");
+        } else {
+            System.out.println("Estas son tus tareas:");
+            for (int i = 0; i < tasks.size(); i++) {
+                System.out.println((i + 1) + ". " + tasks.get(i));
+            }
+        }
+    }
```
Comment (Frozennfishh) [inline src/main/java/elchino/tasks/TaskList.java]: Maybe this can be under UI, but based on preference

File: src/main/java/elchino/tasks/Task.java [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    /**
+     * Converts the task into a string format for storage.
+     * @return The string format for storage.
+     */
+    public abstract String storeTask();
+
+    @Override
+    public String toString() {
+        return String.format("[%s] %s", isDone ? "X" : " ", description);
+    }
```
Comment (Frozennfishh) [inline src/main/java/elchino/tasks/Task.java:78]: The format of these 2 methods are very similar, maybe can streamline into 1 to avoid confusion when reading and writing into files vs for printing?

File: src/test/java/elchino/tasks/TaskListTest.java [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+        // Test adding 2 tasks
+        taskList.addTask(task2);
+        assertEquals(2, taskList.getSize(), "TaskList should have 2 tasks after adding 2 tasks");
+
+        // Verify that first task is task1
+        assertEquals("Task 1", taskList.getTask(1).getDescription(), "First task should match task1");
+
+        // Verify that second task is task2
+        assertEquals("Task 2", taskList.getTask(2).getDescription(), "Second task should match task2");
```
Comment (Frozennfishh) [inline src/test/java/elchino/tasks/TaskListTest.java:25]: Can be more rigorous in testing, maybe adding more edge cases (mismatches, null lists etc.)

File: src/test/java/elchino/parser/ParserTest.java [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+import elchino.exceptions.ElchinoException;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+
+public class ParserTest {
+    @Test
+    public void testParseEmptyInput() throws ElchinoException {
+        Command command = Parser.parse("");
+        assertInstanceOf(InvalidCommand.class, command, "Empty input should return InvalidCommand");
+    }
```
Comment (Frozennfishh) [inline src/test/java/elchino/parser/ParserTest.java:13]: Perhaps parser test should also test invalid inputs/different input cases

**Review by Frozennfishh — COMMENTED** [review]
LGTM!
Spanish isn't my first language so it threw me off a bit at first, but structure of the code is good with no coding standard violations!
A few nits on import ordering and potential streamlining of certain methods being in certain classes but overall very coherent 👍

File: src/main/java/elchino/ui/Ui.java (hunk 3 above)
Comment (limshaoqi) [reply src/main/java/elchino/ui/Ui.java:58]: yea

File: src/main/java/elchino/data/tasks.txt
Comment (limshaoqi) [reply src/main/java/elchino/data/tasks.txt:1]: bet

File: src/main/java/elchino/Elchino.java (hunk 1 above)
Comment (limshaoqi) [reply src/main/java/elchino/Elchino.java]: vale

File: src/test/java/elchino/tasks/TaskListTest.java (hunk 10 above)
Comment (limshaoqi) [reply src/test/java/elchino/tasks/TaskListTest.java:25]: laze
